### ISSUE TITLE
how to know deployment status of a webapp

### ISSUE THREAD

**Issue by LiquidITGuy** [caprover/caprover#688]
**Is your feature request related to a problem? Please describe.**
Yes, during my ci, I want my gitlab to launch all required things (test, lint, deploy, publishing version on log monitoring, launching e2e automated testing, performance and accessibility testing, mattermost message...) but now there is no (easy?) way to know if the app is really deployed. We can just add a postpone timer before launching front end env testing (or check the cli to know how to do it?)

**Describe the solution you'd like**
Having 2 endpoint, one asynchronous (like this one) and the second synchronous (waiting that the deploy is over (it can be done with a accept header) (if option 1 : answer 202) (if option 2: answer 200 after deployment)

**Describe alternatives you've considered**
adding answer to the webhook with a hypermedia link to the "deployment" ressource with id ({id: ..., next: 'http://fqdn/deployements/:id'} for exemple)
The client have to call this api on loop to know if the deployment is pending, done or failed

For the two solution a clear documentation should be required

**Additional context**
the next link is inspired by level 3 rest maturity model and could be realy usefull

**Comment by githubsaturn** [comment]
This is something that was previously discussed. We'll consider implement this in parallel with the webhook using the same token.

**Comment by r-chris** [comment]
@LiquidITGuy just so that I understand your request correctly: Are you asking for some sort of `healthcheck` that CapRover would monitor (e.g. via custom `http` endpoint: `/health`)?

Would this be similar to what CloudFoundry does here: https://docs.cloudfoundry.org/devguide/deploy-apps/healthchecks.html ?

I assume this request is similar in nature to #503 ?

**Comment by githubsaturn** [comment]
@r-chris no. This is more towards the status of the build process itself. Since webhook response doesn't indicate whether the image building was successful or not, we need an extra endpoint to check the build status

**Comment by r-chris** [comment]
Thanks, got it!

**Comment by githubsaturn** [comment]
Closing this issue in favor of https://github.com/caprover/caprover/issues/727 as it's a more generic solution to this problem
